File: leetcode299.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
    int countA, countB;
		countA = 0;
		countB = 0;
		int secret1[10] = {0};
		int guess1[10] = {0};
		for (int i = 0; i < secret.length(); ++i)
		{
			if (secret[i] == guess[i])
			{
				countA++;
			}
			else
			{
				secret1[secret[i]-'0']++;
				guess1[guess[i]-'0']++;
			}
		}
		for (int i = 0; i < 10; ++i)
		{
			countB += min(secret1[i],guess1[i]);
		}
		return to_string(countA) + 'A' + to_string(countB) + 'B';

    }
};

File: leetcode299.java
public class Solution {
    public String getHint(String secret, String guess) {
        int bull = 0, cow = 0;
        int[] arr1 = new int[10];
        int[] arr2 = new int[10];
        for(int i = 0; i < secret.length(); i++){
            if(secret.charAt(i) == guess.charAt(i)){
                bull++;
            }
            else{
                arr1[secret.charAt(i) - '0']++;
                arr2[guess.charAt(i) - '0']++;
            }
        }
        for(int i = 0; i < 10; i++){
            cow += Math.min(arr1[i],arr2[i]);
        }
        return (bull+"A"+cow+"B");
    }
}
